runbotics/runbotics-orchestrator-ui/src/src-app/hooks/useProcessImport.ts

```diff
@@ -0,0 +1,75 @@
+import { isValidJson } from '#src-app/views/action/ActionDetails';
+import { AdditionalInfo } from '#src-app/views/process/ProcessBuildView/Modeler/BpmnModeler';
+
+interface ExtractImportInfoProps extends AdditionalInfo {
+    definition: string;
+}
+const useProcessImport = () => {
+    // eslint-disable-next-line complexity
+    const extractImportInfo = (xmlString: string): ExtractImportInfoProps => {
+        const parser = new DOMParser();
+        const serializer = new XMLSerializer();
+        const parsed = parser.parseFromString(xmlString, 'text/xml');
+        const parent = parsed.getElementsByTagName('bpmn2:definitions')[0];
+        const runboticsInfoElement = parsed.getElementsByTagName(
+            'runbotics:additionalInfo'
+        )[0];
+        if (!runboticsInfoElement) {
+            return { definition: xmlString };
+        }
+
+        const importInfo: ExtractImportInfoProps = {
+            definition: '',
+        };
+
+        const executionInfo = parsed.getElementsByTagName(
+            'runbotics:executionInfo'
+        )[0];
+
+        if (executionInfo) {
+            importInfo.executionInfo = executionInfo?.innerHTML.trim() ?? '';
+        }
+
+        const system = parsed.getElementsByTagName('runbotics:system')[0];
```

Comment:
isn't there other way than to use [0]? If it's used so many times than maybe extract 0 to const

Reply:
maybe create a function put in .utils.ts something like:

const getFirstElementByTag = (tagName) => {
const elements = ...
return elements[0]